api/src/main/java/org/pmiops/workbench/cohorts/CohortFactoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package org.pmiops.workbench.cohorts;
+
+import org.pmiops.workbench.db.model.Cohort;
+import org.pmiops.workbench.db.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.Clock;
+
+@Service
+public class CohortFactoryImpl implements CohortFactory {
+
+    private final Clock clock;
+
+    @Autowired
+    public CohortFactoryImpl(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public Cohort createCohort(org.pmiops.workbench.model.Cohort apiCohort, User creator, long workspaceId) {
+        return createCohort(
+                apiCohort.getDescription(),
+                apiCohort.getName(),
+                apiCohort.getType(),
+                apiCohort.getCriteria(),
+                creator,
+                workspaceId
+        );
+    }
+
+    @Override
+    public Cohort duplicateCohort(Cohort original, User creator) {
```

Comment:
so I read through this a bit and I think it might be tough to reuse since the use cases here and with the "Duplicate" feature are a bit different.

It seems like `cloneCohortAndReviews` is more like deep cloning the entire Cohort object into a new workspace whereas `duplicateCohort` is copying a Cohort (without review sets) within its current current workspace.

The functional differences are...
- the name is changed with `duplicateCohort` and not with `cloneCohort`
- `creator` is copied from the target workspace in `cloneCohort`, passed in as an argument for `duplicateCohort`
-  `lastModifiedTime` is copied from the target workspace in `cloneCohort`, `now()` in `duplicateCohort`
-  `setCreationTime` is copied from the target workspace in `cloneCohort`, `now()` in `duplicateCohort`
- `CohortReview` is not copied over in `duplicateCohort`, no annotations either


That said, if we can settle on a single source for `creator/lastModifiedTime/setCreationTime` when being copied over, I think I can refactor the clone functionality to look something like the following

`cloneCohort(String newName, Cohort original, Workspace targetWorkspace)`
`cloneCohortWithReviewSets(...)`

Reply:
In cloneWorkspace, creator/lastModifiedTime/creationTime actually match the behavior in this PR, it's just coded a bit unintuitively (targetWorkspace is always a freshly created workspace by the cloner), so those changes are not a problem.

Another issue we'd need to deal with is the ID copying in CloneWorkspace - i.e. the current method assumes its being cloned into a fresh workspace (which is not a great separation of concerns). I think these are different enough that we may not need to hit the same code paths right now. That said, if there's any code which would make sense to move into `CohortService` (e.g. `cloneCohort` with a `save()`?) and comments which clearly explain the differences, that would be very useful going forward. That may be what you are suggesting.

One other thing I'll note is that I'm not convinced on the permanence of this decision to not copy cohort reviews. I could easily imagine this become a feature request (e.g. optional checkbox when cloning).

Reply:
What IDs are being copied that you are worried about? the annotation copy methods?

Reply:
Actually ignore that, I was thinking ID was copied but that wouldn't work since it's the primary key. It's just the name that gets copied, which you noted.